src/libraries/Common/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -84,16 +82,36 @@ private void ForceSetKeySize(int newKeySize)
         public override byte[] SignHash(byte[] hash)
         {
             ArgumentNullException.ThrowIfNull(hash);
-
             ThrowIfDisposed();
-            SafeEcKeyHandle key = _key.Value;
-            int signatureLength = Interop.Crypto.EcDsaSize(key);
 
-            Span<byte> signDestination = stackalloc byte[SignatureStackBufSize];
-            ReadOnlySpan<byte> derSignature = SignHash(hash, signDestination, signatureLength, key);
+            // We need to duplicate key handle in case it's being used by multiple threads and one of them disposes it
+            using (SafeEvpPKeyHandle key = _key.Value.DuplicateHandle())
+            using (SafeEvpPKeyCtxHandle ctx = Interop.Crypto.EvpPKeyCtxCreate(key))
+            {
+                Interop.Crypto.EvpPKeyCtxConfigureForECDSASign(ctx);
+
+                if (!Interop.Crypto.TryEvpPKeyCtxSignatureSize(ctx, hash, out int sufficientDerSignatureSize))
+                {
+                    throw new CryptographicException();
+                }
+
+                Span<byte> derSignature = sufficientDerSignatureSize <= SignatureStackBufSize ? stackalloc byte[sufficientDerSignatureSize] : new byte[sufficientDerSignatureSize];
```

Comment:
Consider
```suggestion
                Span<byte> derSignature = (uint)sufficientDerSignatureSize <= SignatureStackBufSize ? stackalloc byte[sufficientDerSignatureSize] : new byte[sufficientDerSignatureSize];
```

If `TryEvpPKeyCtxSignatureSize` returns a negative value for whatever reason we'll blow the stack limit.

In general I am very wary of anything other than a `const` going in to `stackalloc`, which is what was happening before.

This feedback applies throughout.

Reply:
this is not present anymore but I've also changed the code to use const stackallocks